src/sender.js

```diff
@@ -161,11 +161,11 @@ class Sender {
      * @return {Promise<boolean>} Resolves to true if there was data in the buffer to send.
      */
     async flush() {
-        const data = this.toBuffer(this.endOfLastRow);
-        if (!data) {
-            return false;
-        }
         return new Promise((resolve, reject) => {
+            const data = this.toBuffer(this.endOfLastRow);
```

Comment:
Does this really protect from the following code?
```js
sender.flush();
sender.flush();
```
?

Promises are constructed synchronously, so `toBuffer()` method should be called twice with no `compact()` call in between. Am I missing anything?